capstone/capapi/views.py

```diff
@@ -56,19 +61,118 @@ class CitationViewSet(viewsets.GenericViewSet, mixins.RetrieveModelMixin, mixins
 
 
 class CaseViewSet(viewsets.GenericViewSet, mixins.RetrieveModelMixin, mixins.ListModelMixin,):
-    """
-    Browse all cases
-    """
     serializer_class = serializers.CaseSerializer
     http_method_names = ['get']
+    lookup_field = 'slug'
     queryset = models.CaseMetadata.objects.all()
     renderer_classes = (renderers.BrowsableAPIRenderer, renderers.JSONRenderer)
-    lookup_field = 'slug'
+    permission_classes = (permissions.IsAPIUser,)
     filter_class = filters.CaseFilter
 
+    def download_many(self):
+        all_cases = self.queryset.all()
+        # get max number of cases to download from query
+        # set max number to daily max if unspecified
+        max_num = int(self.request.query_params.get('max', settings.API_CASE_DAILY_ALLOWANCE))
+
+        try:
```

Comment:
- What does it mean if this raises TypeError? Maybe comment.
- Can simplify by moving the .order_by up to `all_cases = self.queryset.order_by('decision_date')`

Reply:
TypeError happened if no query_params were passed, setting filters to None. Filtering on None caused an error. I think all of this can be simplified by checking whether filters exist in the first place, and only then filtering the cases.